contributors/design-proposals/daemonset-update.md

```diff
@@ -158,9 +173,10 @@ For each pending DaemonSet updates, it will:
 1. Check `DaemonSetUpdateStrategy`:
    - If `OnDelete`: do nothing
    - If `RollingUpdate`:
-     - Compare spec of the daemon pods from step 1 with DaemonSet
-       `.spec.template.spec` to see if DaemonSet spec has changed.
-     - If DaemonSet spec has changed, compare `MaxUnavailable` with DaemonSet
+     - Find all daemon pods that belong to this DaemonSet using label selectors.
+       - Pods with "pod-template-generation" value equal to the DaemonSet's
```

Comment:
Did you mean the pre-1.6 pods? Since we don't have a way to compare pod spec against DaemonSet template, we can only kill them all (they'll get updated with right labels) or ignore them (no update). Killing them is probably not safe. Ignoring them means users will need to manually kill pre-1.6 pods to enable DaemonSet rolling update (if the DaemonSet was created before 1.6).